<?xml version="1.0"?>
<launch>
  <arg name="ID"/>
  <arg name="vehicle"/>
  <arg name="model_config_file"/>
  <arg name="udp_offboard_port_remote"/>
  <arg name="udp_offboard_port_local"/>
  <arg name="mavlink_tcp_port"/>
  <arg name="mavlink_udp_port"/>
  <arg name="fcu_url"/>
  <arg name="x"/>
  <arg name="y"/>
  <arg name="z"/>
  <arg name="heading"/>
  
  <arg name="roll" default="0"/>
  <arg name="pitch" default="0"/>
  <arg name="est" default="ekf2"/>

  <!-- UAV namespace -->
  <group ns="uav$(arg ID)">

    <!-- PX4 params -->
    <env name="PX4_SIM_MODEL" value="$(arg vehicle)" />
    <env name="PX4_ESTIMATOR" value="$(arg est)" />

    <!-- spawn vehicle -->
    <arg name="cmd" value="$(find xacro)/xacro $(find mrs_simulation)/models/mrs_robots_description/urdf/$(arg vehicle).xacro mrs_robots_description_dir:=$(find mrs_simulation)/models/mrs_robots_description 
      namespace:=uav$(arg ID)
      mavlink_udp_port:=$(arg mavlink_udp_port)
      mavlink_tcp_port:=$(arg mavlink_tcp_port)
      model_config_file:=$(arg model_config_file)"/>
    <param name="robot_description" command="$(arg cmd)"/>
    <node name="uav$(arg ID)_spawn" output="screen" pkg="gazebo_ros" type="spawn_model" args="-urdf -param robot_description -model uav$(arg ID) -package_to_model -x $(arg x) -y $(arg y) -z $(arg z) -R $(arg roll) -P $(arg pitch) -Y $(arg heading)"/>
  
  </group>
</launch>
